src/utils.js

```diff
@@ -287,6 +303,40 @@ genotet.utils.keySet = function(collection) {
 };
 
 /**
+ * Validates email address.
+ * @param {string} email
+ * @return {boolean}
+ */
+genotet.utils.validateEmail = function(email) {
```

Comment:
The caller in this case genotet.user. "utils", which indicates a general-purposed utitlity set, shall provide util functions for multiple callers, rather than a specific caller. In this case, genotet.user is a specific caller. Therefore it is arguably not appropriate to have a util function designed for a specific caller.

On the other hand, utils.validateRegex is a general-purpose util function that can be used by any caller. So you can write that in utils.js.

What you argued is possibly true, in terms of the caller would not want to care about the rules of a username being valid (i.e. the regex). In this case, it should be the caller (genotet.user) that provides a wrapper function to the util's general-purposed function, such as
```js
// user.js
genotet.user.validateUsername = function(username) {
  return genotet.utils.validateRegex(username, genotet.user.USERNAME_REGEX_);
}

// utils.js
genotet.utils.validateRegex = function(str, regex) {
  // ...
};
```